### PR TITLE
Added Rancher <-> Ubuntu k8s integration documentation

### PR DESCRIPTION
**Pull request by CalvinHartwell**
- Fixed a typo from the installation commands where sudo is not required. 
- Added third-party integrations section which will be increasing in size soon. 
- Added Rancher integration documentation. 

_**Unticked and Re-ticked Allow edits from maintainers as requested**_

### CODE REVIEW

**Comment by CalvinHartwell** [comment]
Hmm, the preview appears to be missing my changes: [https://kubernetes.io/docs/getting-started-guides/ubuntu/](https://kubernetes.io/docs/getting-started-guides/ubuntu/) - any ideas?

**Comment by CalvinHartwell** [comment]
Preview looks good to me. 
[https://deploy-preview-7607--kubernetes-io-master-staging.netlify.com/docs/getting-started-guides/ubuntu/](https://deploy-preview-7607--kubernetes-io-master-staging.netlify.com/docs/getting-started-guides/ubuntu/)

[https://deploy-preview-7607--kubernetes-io-master-staging.netlify.com/docs/getting-started-guides/ubuntu/rancher/](https://deploy-preview-7607--kubernetes-io-master-staging.netlify.com/docs/getting-started-guides/ubuntu/rancher/)

File: docs/getting-started-guides/ubuntu/rancher.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+{% capture overview %}
+This repository explains how to deploy Rancher 2.0alpha on Canonical Kubernetes. 
+
+These steps are currently in alpha/testing phase and will most likely change. 
+
+The original documentation for this integration can be found here [https://github.com/CalvinHartwell/canonical-kubernetes-rancher/](https://github.com/CalvinHartwell/canonical-kubernetes-rancher/). 
+
+{% endcapture %}
+{% capture prerequisites %}
+This page assumes you have a working Juju deployed Canonical Kubernetes Cluster.
```
Comment (heckj) [inline docs/getting-started-guides/ubuntu/rancher.md]: Wording recommendation:
> To use this guide, you must have a working kubernetes cluster that was deployed using Canonical's Juju.

File: docs/getting-started-guides/ubuntu/rancher.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+{% endcapture %}
+{% capture prerequisites %}
+This page assumes you have a working Juju deployed Canonical Kubernetes Cluster. 
+{% endcapture %}
+
+
+{% capture steps %}
+## Implementation
+
+To deploy Rancher, we just need to run the Rancher container workload on-top of Kubernetes. Rancher provides their containers through dockerhub ([https://hub.docker.com/r/rancher/server/tags/](https://hub.docker.com/r/rancher/server/tags/)) and can be downloaded freely from the internet. If you're running your own registry or have an offline deployment, the container should be downloaded and pushed to the private registry.
```
Comment (heckj) [inline docs/getting-started-guides/ubuntu/rancher.md]: Wording suggestion to make this more direct:
> You may deploy Rancher on top of Kubernetes. 

In the last sentence in the paragraph, you reference "the private registry" as though it's been spoken of already, but i didn't see it referenced. Would better phrasing be "a private registry."

File: docs/getting-started-guides/ubuntu/rancher.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+
+{% capture steps %}
+## Implementation
+
+To deploy Rancher, we just need to run the Rancher container workload on-top of Kubernetes. Rancher provides their containers through dockerhub ([https://hub.docker.com/r/rancher/server/tags/](https://hub.docker.com/r/rancher/server/tags/)) and can be downloaded freely from the internet. If you're running your own registry or have an offline deployment, the container should be downloaded and pushed to the private registry.  
+
+## Deploying Rancher
+
+To deploy Rancher, we just need to run the Rancher container workload on-top of Kubernetes. Rancher provides their containers through dockerhub ([https://hub.docker.com/r/rancher/server/tags/](https://hub.docker.com/r/rancher/server/tags/)) and can be downloaded freely from the internet. If you're running your own registry or have an offline deployment, the container should be downloaded and pushed to the private registry.
```
Comment (heckj) [inline docs/getting-started-guides/ubuntu/rancher.md]: This whole paragraph is duplicated, you really only need it once - either here or in the prereq. section

File: docs/getting-started-guides/ubuntu/rancher.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+To deploy Rancher, we just need to run the Rancher container workload on-top of Kubernetes. Rancher provides their containers through dockerhub ([https://hub.docker.com/r/rancher/server/tags/](https://hub.docker.com/r/rancher/server/tags/)) and can be downloaded freely from the internet. If you're running your own registry or have an offline deployment, the container should be downloaded and pushed to the private registry.  
+
+## Deploying Rancher
+
+To deploy Rancher, we just need to run the Rancher container workload on-top of Kubernetes. Rancher provides their containers through dockerhub ([https://hub.docker.com/r/rancher/server/tags/](https://hub.docker.com/r/rancher/server/tags/)) and can be downloaded freely from the internet. If you're running your own registry or have an offline deployment, the container should be downloaded and pushed to the private registry.  
+
+### Deploying Rancher with a nodeport
+
+First create a yaml file which will deploy Rancher on kubernetes, save the file as [cdk-rancher-nodeport.yaml](https://raw.githubusercontent.com/CalvinHartwell/canonical-kubernetes-rancher/master/cdk-rancher-nodeport.yaml):
```
Comment (heckj) [inline docs/getting-started-guides/ubuntu/rancher.md]: wording suggestion; change:
> create a yaml file which will deploy Rancher

to:
> create a yaml file that defines how to deploy Rancher

Also, you probably don't need to reference the YAML both at a remote github site and include it below. If you include it with this PR as an inline yaml, our CI system will validate it, and you can reference it directly using {% include %} to have it display visually as well.

File: docs/getting-started-guides/ubuntu/rancher.md [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+If you need to make any changes to the kubernetes configuration file, edit the yaml file and then just use apply again: 
+
+```
+  kubectl apply -f cdk-rancher-nodeport.yaml
+```
+
+### Removing Rancher
+
+If you wish to remove rancher from the cluster, we can do it using kubectl. Deleting constructs in Kubernetes is as simple as creating them:
```
Comment (heckj) [inline docs/getting-started-guides/ubuntu/rancher.md]: wording suggestion; change:
> If you wish to remove rancher from the cluster, we can do it using kubectl.

to:
> You can remove Rancher from your cluster using kubectl.

**Comment by heckj** [comment]
Hi @CalvinHartwell - thank you for the updates! Adding the additional file means we need it included in some table of contents files as well, which is what the CI system is reporting a failure on:

> Error: docs/getting-started-guides/ubuntu/rancher.md doesn't have an entry in the table of contents under _data/*.yml
> Found 1 files without entries. For how to fix it, see http://kubernetes.io/docs/home/contribute/write-new-topic/#creating-an-entry-in-the-table-of-contents

We'll need that updated before we can merge the PR.

/assign

File: docs/getting-started-guides/ubuntu/rancher.md [hunk 6]
```diff
@@ -0,0 +1,198 @@
+---
+title: Rancher Integration with Ubuntu Kubernetes
+---
+
+{% capture overview %}
+This repository explains how to deploy Rancher 2.0alpha on Canonical Kubernetes. 
+
+These steps are currently in alpha/testing phase and will most likely change. 
+
+The original documentation for this integration can be found here [https://github.com/CalvinHartwell/canonical-kubernetes-rancher/](https://github.com/CalvinHartwell/canonical-kubernetes-rancher/).
```
[the remark this answers is not in the document]
Comment (CalvinHartwell) [reply docs/getting-started-guides/ubuntu/rancher.md]: Done, this language actually appears elsewhere in the docs, so I will correct it in my next PR later next week.

File: docs/getting-started-guides/ubuntu/rancher.md (hunk 1 above)
Comment (CalvinHartwell) [reply docs/getting-started-guides/ubuntu/rancher.md]: Fixed.

File: docs/getting-started-guides/ubuntu/rancher.md (hunk 2 above)
Comment (CalvinHartwell) [reply docs/getting-started-guides/ubuntu/rancher.md]: Fixed.

File: docs/getting-started-guides/ubuntu/rancher.md (hunk 3 above)
Comment (CalvinHartwell) [reply docs/getting-started-guides/ubuntu/rancher.md]: Fixed, my bad.

File: docs/getting-started-guides/ubuntu/rancher.md (hunk 4 above)
Comment (CalvinHartwell) [reply docs/getting-started-guides/ubuntu/rancher.md]: Fixed.

File: docs/getting-started-guides/ubuntu/rancher.md (hunk 5 above)
Comment (CalvinHartwell) [reply docs/getting-started-guides/ubuntu/rancher.md]: Fixed.

**Comment by CalvinHartwell** [comment]
@heckj thanks for the feedback, I've amended the doc. I also added another example of using an ingress rule, any chance you could check it out? 

Cheers

**Comment by CalvinHartwell** [comment]
@heckj whoops, I just noticed I missed something, apologies.

**Pull request merged**